examples/rust/digital_signature/core/BUILD.bazel

```diff
@@ -1,37 +1,35 @@
-load("//bazel/rules/risc0:defs.bzl", "risc0_rust_method")
-load("@rules_rust//rust:defs.bzl", "rust_library", "rust_test")
+load("//bazel/rules/risc0:defs.bzl", "risc0_rust_library_pair", "risc0_rust_method")
+load("@rules_rust//rust:defs.bzl", "rust_test")
 
 risc0_rust_method(
     name = "sign",
     srcs = ["src/bin/sign.rs"],
-    rustc_flags = [
-        "-Coverflow-checks=on",
-    ],
+    rustc_flags = ["-Coverflow-checks=on"],
     visibility = ["//visibility:public"],
     deps = [
-        ":core",
+        ":core_guest",
         "//risc0/zkvm/sdk/rust/guest",
     ],
 )
 
-rust_library(
+risc0_rust_library_pair(
     name = "core",
     srcs = ["src/lib.rs"],
     crate_name = "digital_signature_core",
-    rustc_flags = [
-        "-Coverflow-checks=on",
+    guest_deps = [
+        "//risc0/zkvm/sdk/rust/core:core_guest",
+        "@crates_guest//:serde",
     ],
-    visibility = ["//visibility:public"],
-    deps = [
-        "//risc0/zkvm/sdk/rust/core",
-        "@crates//:serde",
+    host_deps = [
+        "//risc0/zkvm/sdk/rust/core:core_host",
+        "@crates_host//:serde",
```

Comment:
I've got a feeling that this will fix my bincode/json woes :+1:

Reply:
Yep that's the plan! :)